be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -199,23 +210,37 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
         RETURN_IF_ERROR(DescriptorTbl::create(obj_pool, t_desc_tbl, &desc_tbl, runtime_state->chunk_size()));
     }
     runtime_state->set_desc_tbl(desc_tbl);
+
+    return Status::OK();
+}
+
+int32_t FragmentExecutor::_calc_dop(const TExecPlanFragmentParams& request) const {
+    int32_t degree_of_parallelism = 1;
+    if (request.__isset.pipeline_dop && request.pipeline_dop > 0) {
+        degree_of_parallelism = request.pipeline_dop;
+    } else {
+        // default dop is a half of the number of hardware threads.
+        degree_of_parallelism = std::max<int32_t>(1, std::thread::hardware_concurrency() / 2);
```

Comment:
Do we not consider `config::pipeline_exec_thread_pool_thread_num`?

Reply:
Good advice, that must be a mistake.

Reply:
Good advice, I'll change it.